Makefile

```diff
@@ -0,0 +1,58 @@
+SRC_DIR := sources
+OBJ_DIR := objects
+
+CXX := g++-9
+
+EXE := a.out
+SRC := $(wildcard $(SRC_DIR)/*.cpp)
+OBJ := $(SRC:$(SRC_DIR)/%.cpp=$(OBJ_DIR)/%.o)
+DOX := doxconf
+
+IO_TEST_DIR		:= io_test
+UNIT_TEST_DIR	:= unit_test
+JSON_FILE_DIR	:= json_files
+
+CXXFLAGS := -std=c++17 -Wall -g
+
+.PHONY: all clean
+
+all: $(EXE)
+
+$(EXE): $(OBJ)
+	$(CXX) $^ -o $@
+
+$(OBJ_DIR)/%.o: $(SRC_DIR)/%.cpp | $(OBJ_DIR)
+	$(CXX) $(CXXFLAGS) -c $< -o $@
```

Comment:
nem vagyok make magus, lehet benezem, de azt nem latom, hogy a header depek itt hogy kerulnek be. Care to explain?

Reply:
ahogy nézem kihagytam néhány flaget